app/components/click_observer_component.rb

```diff
@@ -1,14 +1,22 @@
 # frozen_string_literal: true
 
 class ClickObserverComponent < BaseComponent
-  attr_reader :event_name, :tag_options
+  attr_reader :event_name, :tag_options, :payload
 
-  def initialize(event_name:, **tag_options)
+  def initialize(event_name:, payload: {}, **tag_options)
     @event_name = event_name
+    if payload.is_a?(Hash)
+      @payload = payload
+    else
+      @payload = {}
+    end
```

Comment:
I'm not sure I follow why this condition is needed? The component tests seem to pass with simple assignment.

```suggestion
    @payload = payload
```

Reply:
I was thinking, for the Javascript side of things, would want to ensure, that its in a valid form or else it will throw a javascript error.